### PR TITLE
[GEOT-6545] Encode does not generate a valid WMTS GetCapabilitiesResponse

### PR DESCRIPTION
**Pull request by singhdan**
* Corrections made to WMTS bindings to the AbstractComplexEMFBinding as everything uses EMF (Eclipse Modelling Framework)
* Changed StyleBinding, DimensionBinding and TileMatrixBinding to extend DescriptionTypeBinding so that they can populate the relevant ows:DescriptionType fields
* Added in code to use LayerTypeBinding wherever a DatasetDescriptionSummary element is encountered. This is done by setting the resolvedElementDeclaration so the the LayerTypeBinding is used. In addition, within Encoder.java, a change has been made to specifically use the ResolvedElementDeclaration when encoding
* XMLCodegen configuration for gt-xsd-ows updated to generate the BasicIdentificationTypeOws11Binding, DescriptionTypeOws11Binding and IdentificationTypeOws11Binding to support the encoding correct of WMTS
* Added in unit test for WMTS encoding
* Various spelling corrections

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [X] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [X] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [X] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer.

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):
- [X] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [X] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [X] PR for bug fixes and small new features are presented as a single commit
- [X] Commit message must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [X] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

**Comment by aaime** [comment]
These are big changes, CLA is required. Want to make sure you're ok with sending one before embarking in a review, could you confirm?

**Comment by singhdan** [comment]
Yes I'm fine with doing a CLA. I'll get one sent over this evening

________________________________

**Comment by singhdan** [comment]
CLA sent to info at osgeo.org mailbox

File: modules/extension/xsd/xsd-ows/src/main/java/org/geotools/ows/bindings/BasicIdentificationTypeOws11Binding.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2002-2020, Open Source Geospatial Foundation (OSGeo)
```
Comment (jodygarnett) [inline modules/extension/xsd/xsd-ows/src/main/java/org/geotools/ows/bindings/BasicIdentificationTypeOws11Binding.java:5]: new files just have (c) 2020

**Comment by aaime** [comment]
Well, all builds are looking fine, let's move on and merge.

**Pull request merged**